scripts/data_prep/convert_text_to_mds.py

```
@@ -0,0 +1,352 @@
+        _, _, output_folder_prefix = parse_uri(output_root)
+        try:
+            with tempfile.TemporaryDirectory() as tmp_dir:
+                done_file = os.path.join(tmp_dir, done_file_name)
+                output_object_store.download_object(
+                    os.path.join(output_folder_prefix, done_file_name),
+                    done_file)
+                with open(done_file) as df:
+                    done_file_contents = df.read().splitlines()
+        except FileNotFoundError:
+            return False
+    else:
+        # Read the local done file
+        done_file = os.path.join(output_root, done_file_name)
+        if not os.path.isfile(done_file):
+            return False
+        with open(done_file) as df:
+            done_file_contents = df.read().splitlines()
+    # Compare the arguments
+    prev_args_str = done_file_contents[0]
+    if prev_args_str != args_str:
+        return False
+
+    # Compare file names
+    prev_names = done_file_contents[1:]
+    if len(prev_names) != len(object_names):
+        return False
+    for idx, prev_name in enumerate(prev_names):
+        if object_names[idx] != prev_name:
+            return False
+    return True
+
+
+def write_done_file(folder: str, file_name: str, args_str: str,
+                    object_names: List[str]):
+    with open(os.path.join(folder, file_name), 'w') as done_file:
+        done_file.write('\n'.join([args_str] + object_names) + '\n')
+
+
+def get_done_file_name() -> str:
+    return '.text_to_mds_conversion_done'
+
+
+def main(
+    tokenizer_name: str,
+    output_folder: str,
+    input_folder: str,
+    concat_tokens: int,
+    eos_text: str,
+    bos_text: str,
+    no_wrap: bool,
+    max_mds_writer_workers: int,
+    compression: str,
+    processes: int,
+    args_str: str,
+    reprocess: bool,
+):
+    done_file_name = get_done_file_name()
+    is_remote_output = is_remote_path(output_folder)
+
+    object_names = get_object_names(input_folder)
```

should we raise an error here if no files are found?